src/ServiceControl/Monitoring/HeartbeatMonitor.cs

```diff
@@ -0,0 +1,53 @@
+namespace ServiceControl.Monitoring
+{
+    using System;
+    using System.Threading;
+
+    class HeartbeatMonitor
+    {
+        volatile RecordedHeartbeat heartbeat = new RecordedHeartbeat(HeartbeatStatus.Unknown, null);
```

Comment:
@mikeminutillo according to https://stackoverflow.com/questions/425132/a-reference-to-a-volatile-field-will-not-be-treated-as-volatile-implications it is fine.